comp/collector/collector/collectorimpl/collector.go

```diff
@@ -220,6 +220,14 @@ func (c *collectorImpl) StopCheck(id checkid.ID) error {
 	// delete check from checks map even if we encounter an error
 	defer c.delete(id)
 
+	// remove the check from the stats map
+	defer expvars.RemoveCheckStats(id)
+
+	stats, found := expvars.CheckStats(id)
+	if found {
+		stats.Cancelling = true
```

Comment:
All of the other stats behind `CheckStats` are updated under a lock in the Add function. 
Do we need to lock around updating this property as well?

Reply:
Good catch. I have updated to use the lock